inference-engine/src/gna_plugin/backend/dnn_components.cpp

```diff
@@ -8,32 +8,60 @@
 #include <ie_common.h>
 #include <legacy/ie_layers.h>
 #include <iomanip>
+#include <details/caseless.hpp>
+#include <layers/gna_copy_layer.hpp>
 #include "backend/dnn_types.h"
 
 #include "dnn_components.hpp"
 
 using namespace GNAPluginNS;
+using namespace GNAPluginNS::backend;
 
-intel_dnn_component_t & backend::DnnComponents::addComponent(const std::string layerName, const std::string layerMetaType) {
-    components.emplace_back(layerName, intel_dnn_component_t());
-    auto &currentComponent = components.back().second;
+intel_dnn_component_t & DnnComponents::addComponent(const std::string layerName, const std::string layerMetaType) {
+    auto isDelayed = InferenceEngine::details::CaselessEq<std::string>()(layerMetaType, DelayedCopyLayerName);
+    delayedOperations += isDelayed ? 1 : 0;
+    components.emplace_back(DnnComponentExtra{layerName, {}, isDelayed});
+    auto &currentComponent = components.back().dnnComponent;
 #ifdef PLOT
-    currentComponent.original_layer_name = components.back().first.c_str();
-    std::cout << "IR layer : " << std::left << std::setw(20) << layerName << " " << layerMetaType << "_" << components.size() - 1 << std::endl;
+    currentComponent.original_layer_name = components.back().name.c_str();
 #endif
+    int execOrder = 0;
+    if (!isDelayed) {
+        execOrder = components.size() - 1 - delayedOperations;
+    } else {
+        // todo: not perfect - propose to create mapping table that will be printed out by extra request
+        execOrder = -delayedOperations;
```

Comment:
Would it result in negative number if there are already delayed operations added?
Shouldn't it be 'components.size() - 1' ?

Reply:
I believe that not delayed operations will have positive numbers (starting from 0), while delayed operations will be in negatives (starting from -1)

Reply:
I thought that it should print the same position as in getExecutionOrder function.